File: src/api/toolConfig.ts
import { invoke } from '@tauri-apps/api/core'
import type { CcSubtitleConfig, ProofreadConfig, TextConversionConfig } from '../types'

export function loadTextConversionConfig() {
  return invoke<TextConversionConfig>('load_text_conversion_config')
}

export function saveTextConversionConfig(config: TextConversionConfig) {
  return invoke<void>('save_text_conversion_config', { config })
}

export function loadProofreadConfig() {
  return invoke<ProofreadConfig>('load_proofread_config')
}

export function saveProofreadConfig(config: ProofreadConfig) {
  return invoke<void>('save_proofread_config', { config })
}

export function loadCcSubtitleConfig() {
  return invoke<CcSubtitleConfig>('load_cc_subtitle_config')
}

export function saveCcSubtitleConfig(config: CcSubtitleConfig) {
  return invoke<void>('save_cc_subtitle_config', { config })
}

File: src-tauri/src/commands/tool_config.rs
use crate::models::tool_config::{CcSubtitleConfig, ProofreadConfig, TextConversionConfig};
use crate::services::tool_config_store;
use tauri::AppHandle;

const TOOLBOX_FILE: &str = "toolbox.yaml";

#[tauri::command]
pub fn load_text_conversion_config(app: AppHandle) -> Result<TextConversionConfig, String> {
    tool_config_store::load_text_conversion(&app, TOOLBOX_FILE)
}

#[tauri::command]
pub fn save_text_conversion_config(
    app: AppHandle,
    config: TextConversionConfig,
) -> Result<(), String> {
    tool_config_store::save_text_conversion(&app, TOOLBOX_FILE, &config)
}

#[tauri::command]
pub fn load_proofread_config(app: AppHandle) -> Result<ProofreadConfig, String> {
    tool_config_store::load_proofread(&app, TOOLBOX_FILE)
}

#[tauri::command]
pub fn save_proofread_config(app: AppHandle, config: ProofreadConfig) -> Result<(), String> {
    tool_config_store::save_proofread(&app, TOOLBOX_FILE, &config)
}

#[tauri::command]
pub fn load_cc_subtitle_config(app: AppHandle) -> Result<CcSubtitleConfig, String> {
    tool_config_store::load_cc_subtitle(&app, TOOLBOX_FILE)
}

#[tauri::command]
pub fn save_cc_subtitle_config(app: AppHandle, config: CcSubtitleConfig) -> Result<(), String> {
    tool_config_store::save_cc_subtitle(&app, TOOLBOX_FILE, &config)
}
